conans/client/rest_client_folder.py

```diff
@@ -0,0 +1,200 @@
+import os
+import sys
+import textwrap
+from distutils.dir_util import copy_tree
+from fnmatch import fnmatch
+from io import StringIO
+
+import yaml
+
+from conan.api.output import ConanOutput
+from conan.internal.cache.home_paths import HomePaths
+from conans.client.cmd.export import cmd_export
+from conans.errors import ConanException, PackageNotFoundException, RecipeNotFoundException
+from conans.model.conf import ConfDefinition
+from conans.model.recipe_ref import RecipeReference
+from conans.util.files import load, save
+
+
+class RestApiClientFolder:
+    """
+    Implements the RestAPI but instead of over HTTP for a remote server, using just
+    a local folder assuming the conan-center-index repo layout
+    """
+
+    def __init__(self, remote):
+        self._remote = remote
+        cache_folder = self._remote.url
+        self._remote_cache_dir = "{}/.cache".format(cache_folder)
+        hook_folder = HomePaths(self._remote_cache_dir).hooks_path
```

Comment:
this is pretty clever, using the same folder as the place to place the extension in!

Reply:
Yes, it is a nested fully fledged cache inside the folder, so we can leverage all the existing "export" code, so putting the hook in the cache was a no-brainer.